src/sys/unix.rs

```diff
@@ -1833,6 +1833,36 @@ impl crate::Socket {
         .map(|_| ())
     }
 
+    /// This method is deprecated, use [`bind_device_by_index_v4`].
+    ///
+    /// [`bind_device_by_index_v4`]: crate::Socket::bind_device_by_index_v4
+    #[cfg(all(
+        feature = "all",
+        any(
+            target_os = "ios",
+            target_os = "macos",
+            target_os = "tvos",
+            target_os = "watchos",
+        )
+    ))]
+    #[cfg_attr(
+        docsrs,
+        doc(cfg(all(
+            feature = "all",
+            any(
+                target_os = "ios",
+                target_os = "macos",
+                target_os = "tvos",
+                target_os = "watchos",
+            )
+        )))
+    )]
+    #[deprecated]
+    pub fn bind_device_by_index(&self, interface: Option<NonZeroU32>) -> io::Result<()> {
+        let index = interface.map_or(0, NonZeroU32::get);
+        unsafe { setsockopt(self.as_raw(), IPPROTO_IP, libc::IP_BOUND_IF, index) }
```

Comment:
Please call `bind_device_by_index_v4` instead to show that's the preferred function.

Reply:
Done!